src/cromshell/counts/command.py

```diff
@@ -0,0 +1,297 @@
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILED
+    else:  # catch all
+        task_status_font = "yellow"
+
+    # Format and print task summary
+    formatted_task_summary = (
+        f"{indent}{call}\t{shards_running} Running, "
+        f"{shards_done} Done, {shards_retried} Preempted, {shards_failed} Failed"
+    )
+
+    print(colored(formatted_task_summary, color=task_status_font))
+
+    # If the task has shards that failed list them
+    if shards_failed:
+        failed_shards_index = get_list_of_failed_shards(shards=shards)
+        if failed_shards_index != [-1]:  # Prints only if task was scattered
+            # Format and print task failed shards
+            failed_shards_summary = f"{indent}Failed shards: {failed_shards_index}"
+            print(colored(failed_shards_summary, color=task_status_font))
+
+    # If unknown status present append its count to print out.
+    if shards_unknown > 0:
+        formatted_task_summary += f", {shards_unknown} Unknown"
+        DelayedLogMessage.save_log_message(
+            log_level=logging.WARNING,
+            log_message="Cromshell found the following unknown task status(es) "
+            f"{unknown_shard_status}, an unknown task status is a status that does "
+            f"not match the following known task statuses:  {TaskStatus.list()} . "
+            "Please report all relevant info to Cromshell git repository so we can "
+            "improve our code. Thank You.",
+        )
+
+
+def print_task_status_summary(workflow_metadata: dict) -> None:
+    """
+    Prints the status count for each task in a workflow.
+    Does NOT run expose subworkflows
+
+    :param workflow_metadata: Metadata of the workflow to process
+    :return:
+    """
+
+    workflow_status_summary = {}
+    for task in workflow_metadata["calls"]:
+        shards = workflow_metadata["calls"][task]
+
+        # Add the status counts for this task to dictionary holding other
+        # task status counts
+        workflow_status_summary[task] = get_shard_status_count(shards)
+
+    io_utils.pretty_print_json(format_json=workflow_status_summary)
+
+
+def get_shard_status_count(shards: list) -> Dict[str, int]:
+    """
+    Count the number of shards for each status type and return as dictionary.
+    :param shards: The metadata for all shards in a scatter or shard of a single task
+    :return:
+    """
+
+    statuses_counts = {}
+    grouped_shards = group_shards_by_status(shards=shards)
+
+    for status in grouped_shards:
+        statuses_counts[status] = len(grouped_shards[status])
+
+    return statuses_counts
+
+
+def get_list_of_failed_shards(shards: list) -> list:
+    """
+    Get a list of the failed shards' indexes
+    :param shards: The metadata for all shards in a scatter or shard of a single task
+    :return:
+    """
+
+    failed_shards_index = []
+    grouped_shards = group_shards_by_status(shards=shards)
+
+    for shard in grouped_shards["Failed"]:
```

Comment:
I just realized one potential bug: this would error out if "Failed" is not in the dictionary, i.e. no failed shards/tasks.

1. use `grouped_shards.get('Failed',list())` instead 
2. a test case should catch that
3.

Reply:
done